scripts/cba/make_indicators.py

```diff
@@ -65,6 +65,61 @@ def calculate_total_system_cost(n):
     }
 
 
+def calculate_co2_emissions_per_carrier(n: pypsa.Network) -> pd.Series:
+    """
+    Calculate total CO2 emissions per carrier in the PyPSA network.
+
+    This function computes the energy balance of the network, aggregates it over the entire time period,
+    and then sums the CO2 emissions grouped by bus_carrier and carrier.
+
+    Then it filters to include only positive CO2 emissions (negative values are carbon sinks).
+
+    Parameters
+    ----------
+    n : pypsa.Network
+        The PyPSA network object for which to calculate CO2 emissions.
+
+    Returns
+    -------
+    pandas.Series
+        A Series containing the total CO2 emissions per carrier, indexed by carrier, with only positive values included.
+    """
+    energy_balance = n.statistics.energy_balance(aggregate_time="sum")
+    CO2_per_carrier = (
+        energy_balance.groupby(["bus_carrier", "carrier"]).sum().loc["co2"]
+    )
+    CO2_emissions_per_carrier = CO2_per_carrier[CO2_per_carrier > 0]
```

Comment:
I was just thinking about this part. So in the CBA methodology this is how they are describing how the CO2 emissions are calculated (based on the generation). But I think they are not modelling CO2 management (DAC for example). What would make sense to me is to apply the societal costs of CO2 emissions to the net emissions (so emissions of generation - emissions captured), so the emissions which really remain in the atmosphere. The net emissions are given in the final snapshot of your co2 atmosphere store. What do you think? Should we change the calculation?